File: Pod/TestClasses/FitnesseTestSuiteXMLResultParser.m
#import "FitnesseTestSuiteXMLResultParser.h"

@interface FitnesseTestSuiteXMLResultParser () <NSXMLParserDelegate>

@property (nonatomic, assign) NSUInteger failedTestSuiteCount;

@end

@implementation FitnesseTestSuiteXMLResultParser

- (BOOL) resultForTestSuiteXMLData:(NSData *) data  {
    
    [self parseTestSuiteXMLData:data];
    
    return self.failedTestSuiteCount == 0;
    
}

//MARK: Parsing Private Helpers

- (void) parseTestSuiteXMLData:(NSData*) data {
    
    NSXMLParser *parser = [[NSXMLParser alloc] initWithData:data];
    
    parser.delegate = self;
    
    [parser parse];
    
}

- (void) parseTestSuiteFailuresForElementAttributes:(NSDictionary*)attributesDict {
    
    NSUInteger failuresCount = [[attributesDict objectForKey:@"failures"] integerValue];
    
    self.failedTestSuiteCount += failuresCount;
    
}

//MARK: NSXMLParserDelegate

- (void)parser:(NSXMLParser *)parser parseErrorOccurred:(NSError *)parseError {
    
    [NSException exceptionWithName:parseError.localizedDescription reason:parseError.localizedFailureReason userInfo:parseError.userInfo];
    
}

- (void)parser:(NSXMLParser *)parser didStartElement:(nonnull NSString *)elementName namespaceURI:(nullable NSString *)namespaceURI qualifiedName:(nullable NSString *)qName attributes:(nonnull NSDictionary<NSString *,NSString *> *)attributeDict {
    
    [self parseTestSuiteFailuresForElementAttributes:attributeDict];
    
}

@end

File: Example/OCSlimProject_FitnesseTests/FitnesseTestSuiteXMLResultParser.swift
import Foundation

class FitnesseTestSuiteXMLResultParser: NSObject, NSXMLParserDelegate {
    
    var failedTestSuiteCount = 0
    
    internal func resultForTestSuiteXMLData(data: NSData) -> Bool {
        parseTestSuiteXMLData(data)
        return failedTestSuiteCount == 0
    }
    
    //MARK: Parsing Helpers
    
    private func parseTestSuiteXMLData(data: NSData) {
        let parser = NSXMLParser(data: data)
        parser.delegate = self
        parser.parse()
    }
    
    private func parseTestSuiteFailuresForElementAttributes(attributes attributeDict: [String: String]) {
        if let failuresCount = attributeDict["failures"] {
            failedTestSuiteCount += Int(failuresCount)!
        }
    }
    
    //MARK: NSXMLParserDelegate
    
    func parser(parser: NSXMLParser, didStartElement elementName: String, namespaceURI: String?, qualifiedName qName: String?, attributes attributeDict: [String : String]) {
        
        parseTestSuiteFailuresForElementAttributes(attributes: attributeDict)
    }
    
    func parser(parser: NSXMLParser, parseErrorOccurred parseError: NSError) {
        
        fatalError(parseError.localizedDescription)
    }
}
